README.md

```diff
@@ -13,4 +13,46 @@ See individual package READMEs for more info. In general, this monorepo depends
 ```bash
 pnpm # @^8.7
 node # @^18
+docker # with docker-compose available
 ```
+
+### Running locally
+
+The server can be started locally from a local rebuild of source code using:
+
+```bash
+pnpm compose:up
+```
+
+This will start the server listening on `localhost:8080` with a service container
+running a fresh Oxigraph graph store backend listening on `locahost:33378`. Note
+that the store will initialize empty, but store its data persistently at `tmp/oxigraph_data`
+so that you can upload data and continue to reuse it between launches of the app.
+
+**Seeding data to the graph store:** You can use the script `scripts/upload_to_local_deployment.ts`,
+which ingests any JSON-LD document from standard in and emits triples to the docker-compose
+Oxigraph backend (requires the composed cluster to be running). You can set the environment variable
+`DM_DEV_OXIGRAPH_URL` to something other than `http://localhost:33378` (the default) if you are trying
+to write to some other Oxigraph location.
+
+**WARNING** because of the [algebraic properties of blank nodes](https://docmaps.knowledgefutures.org/pub/eqb8u4v0/release/2),
+SPARQL Update protocol does not support idempotent writes. If you upload the same document multiple times,
+all blank nodes and associated triples will be presumed unique in relation to previous uploads
+of "similar seeming" blank nodes, and will result in duplication in the dataset. Named nodes will
+not be duplicated. For example, Steps and Actions are usually blank nodes, but Docmaps have ids.
+This means that if you upload a typical json-ld Docmap twice, all the steps will be doubled. If an
+action has an ID, then every copy of a step will point to it, whereas if not, there will be a copy
+of the action.
+
+To upload a JSONLD file:
+
+```bash
+<my_docmap.jsonld npx tsx scripts/upload_to_local_deployment.ts
```

Comment:
I'm a little thrown off by this syntax—my bash is a bit rusty. By `<my_docmap.jsonld` here, are you saying I should literally do exactly this syntax, replacing `my_docmap.jsonld` with the exact filename of my docmap? 

e.g. `</tmp/docmaps/andrews_docmap.jsonld`? Or do I need to cat the contents of my docmap and then pipe it to `npx tsx scripts/upload_to_local_deployment.ts`?

Reply:
Answering my own question, it looks like this command worked:

`</tmp/a-docmap.jsonld npx tsx scripts/upload_to_local_deployment.ts`

Reply:
@andrewedstrom , `</some/fd script` is the same stdin redirect you are accustomed to from the format `script < /some/fd` and is therefore effectively the same as `cat /some/fd | script` except without the extra process spawn :) i conventionally put it at the beginning so it reads left-to-right like a pipe but could change it.

Reply:
Fascinating! I slightly prefer writing it `script < /some/fd` but that's really just a matter of what I happen to be familiar with.